File: scripts/py-compile-suite/inputs/29_fprintf_with_transpose.m
% The tricky case: fprintf with format specifiers + transpose in same statement
A = [1 2 3; 4 5 6];
v = [10; 20; 30];
fprintf(1, 'A size: %d x %d\n', size(A));
fprintf('column vector: %s\n', mat2str(v));
row = v';
fprintf('row vector: %s\n', mat2str(row));
B = A';
fprintf('transposed A has %d rows\n', size(B, 1));
result = A*v;
fprintf('product = [%d %d]\n', result(1), result(2));

File: scripts/py-compile-suite/outputs/29_fprintf_with_transpose.py
import numpy as np

# The tricky case: fprintf with format specifiers + transpose in same statement
A = np.array([[1, 2, 3], [4, 5, 6]])
v = np.array([[10], [20], [30]])
print('A size: %d x %d\n', A.shape)
print('column vector: %s' % (mat2str(v),))
row = v.T
print('row vector: %s' % (mat2str(row),))
B = A.T
print('transposed A has %d rows' % (B.shape[0],))
result = A*v
print('product = [%d, %d]' % (result[0], result[1]))
